internal/cmd/plugin/color.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright The CloudNativePG Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/logrusorgru/aurora/v4"
+	"github.com/spf13/cobra"
+	"golang.org/x/term"
+)
+
+type colorValue string
+
+const (
+	// colorAlways is meant to output colorized output always
+	colorAlways colorValue = "always"
+	// colorAuto is meant to output colorized output only when the output is attached to a terminal
+	colorAuto colorValue = "auto"
+	// colorNever is meant to output colorized output never
+	colorNever colorValue = "never"
+)
+
+// String implements pflag.Value interface
+func (e colorValue) String() string {
+	return string(e)
+}
+
+// Set implements pflag.Value interface
+func (e *colorValue) Set(val string) error {
+	colorVal := colorValue(val)
+	if colorVal != colorAlways && colorVal != colorAuto && colorVal != colorNever {
+		return fmt.Errorf("should be one of 'always', 'auto', or 'never'")
+	}
+	*e = colorVal
+	return nil
+}
+
+// Type implements pflag.Value interface
+func (e *colorValue) Type() string {
+	return "string"
+}
+
+// ConfigureColor renews aurora.DefaultColorizer based on flags and TTY
+func ConfigureColor(cmd *cobra.Command) {
+	configureColor(cmd, term.IsTerminal(int(os.Stdout.Fd())))
```

Comment:
The library function used, `term.IsTerminal`, seems pretty coarse, and is returning true on code that cannot support colorizing sequences (eg the Plan9 tools).
That said, it discriminates when using pipes, so it’s somewhat useful.